scriptworker/test/test_cot_verify.py

```diff
@@ -1100,20 +1089,167 @@ def test_verify_link_in_task_graph_fuzzy_match_exception(chain, decision_link, b
+        return {"pushes": pushes}
+
+    mocker.patch.object(cotverify, 'load_json_or_yaml_from_url', new=fake_load)
+    assert await cotverify.get_pushlog_info(decision_link) == {"pushes": pushes}
+
+
+# verify_parent_task_definition {{{1
+@pytest.mark.asyncio
+@pytest.mark.parametrize("name,task_id,path,decision_task_id,decision_path", ((
+    "decision", "VQU9QMO4Teq7zr91FhBusg", os.path.join(COTV2_DIR, "decision_hg-push.json"),
+    "VQU9QMO4Teq7zr91FhBusg", os.path.join(COTV2_DIR, "decision_hg-push.json"),
+), (
+    "action", "MP8uhRdMTjm__Q_sA0GTnA", os.path.join(COTV2_DIR, "action_relpro.json"),
+    "VQU9QMO4Teq7zr91FhBusg", os.path.join(COTV2_DIR, "decision_hg-push.json"),
+), (
+    "decision", "D4euZNyCRtuBci-fnsfn7A", os.path.join(COTV2_DIR, "cron.json"),
+    "D4euZNyCRtuBci-fnsfn7A", os.path.join(COTV2_DIR, "cron.json"),
+)))
+async def test_verify_parent_task_definition(chain, name, task_id, path,
+                                             decision_task_id, decision_path, mocker):
+    link = cotverify.LinkOfTrust(chain.context, name, task_id)
+    link.task = load_json_or_yaml(path, is_path=True)
+    if task_id == decision_task_id:
+        decision_link = link
     else:
-        cotverify.verify_firefox_decision_command(decision_link)
+        decision_link = cotverify.LinkOfTrust(chain.context, 'decision', decision_task_id)
+        decision_link.task = load_json_or_yaml(decision_path, is_path=True)
+
+    mocker.patch.object(cotverify, 'load_json_or_yaml_from_url', new=cotv2_load_url)
+    mocker.patch.object(cotverify, 'load_json_or_yaml', new=cotv2_load)
+    mocker.patch.object(cotverify, 'get_pushlog_info', new=cotv2_pushlog)
+
+    chain.links = list(set([decision_link, link]))
+    await cotverify.verify_parent_task_definition(
+        chain, link
+    )
+
+
+@pytest.mark.asyncio
+async def test_verify_parent_task_definition_bad_project(chain, mocker):
+    link = cotverify.LinkOfTrust(chain.context, 'decision', "VQU9QMO4Teq7zr91FhBusg")
+    link.task = load_json_or_yaml(os.path.join(COTV2_DIR, "decision_hg-push.json"), is_path=True)
+
+    def fake_url(*args):
+        return "https://fake_server"
+
+    mocker.patch.object(cotverify, 'load_json_or_yaml_from_url', new=cotv2_load_url)
+    mocker.patch.object(cotverify, 'load_json_or_yaml', new=cotv2_load)
+    mocker.patch.object(cotverify, 'get_pushlog_info', new=cotv2_pushlog)
+    mocker.patch.object(cotverify, 'get_source_url', new=fake_url)
+
+    chain.links = [link]
+    with pytest.raises(CoTError):
+        await cotverify.verify_parent_task_definition(
+            chain, link
+        )
+
+
+@pytest.mark.asyncio
+async def test_verify_parent_task_definition_bad_comment(chain, mocker):
+    link = cotverify.LinkOfTrust(chain.context, 'decision', "VQU9QMO4Teq7zr91FhBusg")
+    link.task = load_json_or_yaml(os.path.join(COTV2_DIR, "decision_hg-push.json"), is_path=True)
+    link.task['payload']['env']['GECKO_COMMIT_MSG'] = "invalid comment"
```

Comment:
Just to make sure I understand, this commit message becomes invalid because it doesn't match the `mach` command, doesn't it?

Reply:
It doesn't match the pushlog's comment (try) or `' '` (non-try). Since this isn't try, it should be `' '`.

Reply:
I have no idea why it's `' '` either -- that's from a ancient comment in mozilla-taskcluster from jlal :)